### PR TITLE
ref(processor) Inject dependencies into envelope processor service

### PR DESCRIPTION
**Pull request by olksdr**
Remove from registry from the `EnvelopeProcessorService`.

ref: #1818 

#skip-changelog

### CODE REVIEW

File: relay-server/src/actors/processor.rs [hunk 1]
```diff
@@ -2907,20 +2930,25 @@ mod tests {
     }
 
     fn create_test_processor(config: Config) -> EnvelopeProcessorService {
+        let (envelope_manager, _) = mock_service("envelope_manager", (), |&mut (), _| {});
+        let (outcome_aggregator, _) = mock_service("outcome_aggregator", (), |&mut (), _| {});
+        let (project_cache, _) = mock_service("project_cache", (), |&mut (), _| {});
+        let (upstream_relay, _) = mock_service("upstream_relay", (), |&mut (), _| {});
```
Comment (jjbayer) [inline relay-server/src/actors/processor.rs:2936]: nit: Should we make the third argument to `mock_service` optional, or create a wrapper `noop_service` or similar that takes only one argument?

File: relay-server/src/actors/processor.rs [hunk 2]
```diff
@@ -2907,20 +2930,25 @@ mod tests {
...
             rate_limiter: None,
             #[cfg(feature = "processing")]
             geoip_lookup: None,
         }
     }
 
-    #[test]
-    #[ignore = "The REGISTRY panics if the Addr of an service (that is not yet started) is queried,
-                hence this test fails. The old SystemRegistry returned dummy Addr's hence this did
-                not fail."]
```
Comment (jjbayer) [inline relay-server/src/actors/processor.rs:2922]: Nice!

File: relay-server/src/actors/processor.rs [hunk 3]
```diff
@@ -538,7 +540,14 @@ pub struct EnvelopeProcessorService {
...
 impl EnvelopeProcessorService {
     /// Creates a multi-threaded envelope processor.
-    pub fn new(config: Arc<Config>, _redis: Option<RedisPool>) -> anyhow::Result<Self> {
+    pub fn new(
+        config: Arc<Config>,
+        _redis: Option<RedisPool>,
+        envelope_manager: Addr<EnvelopeManager>,
+        outcome_aggregator: Addr<TrackOutcome>,
+        project_cache: Addr<ProjectCache>,
+        upstream_relay: Addr<UpstreamRelay>,
```
Comment (jjbayer) [inline relay-server/src/actors/processor.rs:549]: nit: We could think about putting these in a separate `Receivers` struct to reduce the number of arguments.

**Review by jjbayer — APPROVED** [review]

File: relay-server/src/actors/processor.rs (hunk 3 above)
Comment (olksdr) [reply relay-server/src/actors/processor.rs:549]: yep, this is in the following PR: https://github.com/getsentry/relay/pull/1947

File: relay-server/src/actors/processor.rs (hunk 1 above)
Comment (olksdr) [reply relay-server/src/actors/processor.rs:2936]: yeah, I will look into this in the follow-ups

**Pull request merged**